src/components/autocomplete/js/autocompleteController.js

```diff
@@ -635,10 +635,11 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
   function fetchResults (searchText) {
     var items = $scope.$parent.$eval(itemExpr),
         term  = searchText.toLowerCase(),
-        isList = angular.isArray(items);
+        isList = angular.isArray(items),
+        isPromise = !!items.then; // Every promise should contain a `then` property
```

Comment:
This is PR is kinda old, but just to throw my 2 cents: Angular has an internal function, called `isPromiseLike` that checks whether an object is a promise like this: `obj && isFunction(obj.then)`, maybe we could use the same?